openmdao/components/linear_system.py

```diff
@@ -85,3 +89,23 @@ def apply_linear(self, params, unknowns, dparams, dunknowns, dresids, mode):
                 dparams['A'] += np.outer(unknowns['x'], dresids['x']).T
             if 'b' in dparams:
                 dparams['b'] -= dresids['x']
+
+    def solve_linear(self, dumat, drmat, vois, mode=None):
+        """ LU backsubstitution to solve the derivatives of the linear system."""
+
+        m = self.size
+
+        if mode == 'fwd':
+            sol_vec, rhs_vec = self.dumat, self.drmat
+            t=0
+        else:
+            sol_vec, rhs_vec = self.drmat, self.dumat
+            t=1
+
+        for voi in vois:
+            rhs = np.zeros((m, ))
```

Comment:
self.size doesn't change during a run does it?  If not, we could cache the rhs to avoid creating a new one every time.